### PR TITLE
various fixes to skip processing (change check, removal from skipped …

### PR DESCRIPTION
**Pull request by gabemontero**
…list) per https://bugzilla.redhat.com/show_bug.cgi?id=1645463

/assign @bparees

### CODE REVIEW

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -207,10 +207,14 @@ func (h *Handler) VariableConfigChanged(srcfg *v1alpha1.SamplesResource, cm *cor
 	// Split will return an item of len 1 if skippedStreams is empty;
 	// verified via testing and godoc of method
 	if len(skippedStreams) > 0 {
+		// gotta trim any remaining space to get array size comparison correct
+		// a spling of say 'jenkins ' will create an array of size 2
```
Comment (bparees) [inline pkg/stub/handler.go]: s/spling/string/ ?

File: pkg/stub/handler.go [hunk 2]
```diff
@@ -207,10 +207,14 @@ func (h *Handler) VariableConfigChanged(srcfg *v1alpha1.SamplesResource, cm *cor
...
+		skippedStreams = strings.TrimSuffix(skippedStreams, " ")
```
Comment (bparees) [inline pkg/stub/handler.go]: Seems like https://golang.org/pkg/strings/#TrimSpace would be better.

File: pkg/stub/handler.go [hunk 3]
```diff
@@ -207,10 +207,14 @@ func (h *Handler) VariableConfigChanged(srcfg *v1alpha1.SamplesResource, cm *cor
...
 		streams = strings.Split(skippedStreams, " ")
 	}
+
 	if len(streams) != len(srcfg.Spec.SkippedImagestreams) {
-		logrus.Printf("SkippedImageStreams changed from %s, processing %v", skippedStreams, srcfg)
+		logrus.Printf("SkippedImageStreams check 1 changed from %s, processing %v", skippedStreams, srcfg)
```
Comment (bparees) [inline pkg/stub/handler.go]: leftover debug?

File: pkg/stub/handler.go [hunk 4]
```diff
@@ -220,7 +224,7 @@ func (h *Handler) VariableConfigChanged(srcfg *v1alpha1.SamplesResource, cm *cor
 	}
 	for _, stream := range srcfg.Spec.SkippedImagestreams {
 		if _, ok := streamsMap[stream]; !ok {
-			logrus.Printf("SkippedImageStreams changed from %s, processing %v", skippedStreams, srcfg)
+			logrus.Printf("SkippedImageStreams check 2 changed from %s, processing %v", skippedStreams, srcfg)
```
Comment (bparees) [inline pkg/stub/handler.go]: debug?

File: pkg/stub/handler.go (hunk 3 above)
Comment (bparees) [reply pkg/stub/handler.go]: (if you're intending to disambiguate the message from the other case, please make it something more human readable)

File: pkg/stub/handler.go [hunk 5]
```diff
@@ -234,10 +238,13 @@ func (h *Handler) VariableConfigChanged(srcfg *v1alpha1.SamplesResource, cm *cor
 	// Split will return an item of len 1 if skippedStreams is empty;
 	// verified via testing and godoc of method
 	if len(skippedTemps) > 0 {
+		// gotta trim any remaining space to get array size comparison correct
+		// a spling of say 'jenkins ' will create an array of size 2
+		skippedTemps = strings.TrimSuffix(skippedTemps, " ")
```
Comment (bparees) [inline pkg/stub/handler.go]: TrimSpace

File: pkg/stub/handler.go [hunk 6]
```diff
@@ -234,10 +238,13 @@ func (h *Handler) VariableConfigChanged(srcfg *v1alpha1.SamplesResource, cm *cor
...
 		temps = strings.Split(skippedTemps, " ")
 	}
 	if len(temps) != len(srcfg.Spec.SkippedTemplates) {
-		logrus.Printf("SkippedTemplates changed from %s, processing %v", skippedTemps, srcfg)
+		logrus.Printf("SkippedTemplates check 1 changed from %s, processing %v", skippedTemps, srcfg)
```
Comment (bparees) [inline pkg/stub/handler.go]: same comment on output

File: pkg/stub/handler.go [hunk 7]
```diff
@@ -247,7 +254,7 @@ func (h *Handler) VariableConfigChanged(srcfg *v1alpha1.SamplesResource, cm *cor
 	}
 	for _, temp := range srcfg.Spec.SkippedTemplates {
 		if _, ok := tempsMap[temp]; !ok {
-			logrus.Printf("SkippedTemplates changed from %s, processing %v", skippedTemps, srcfg)
+			logrus.Printf("SkippedTemplates check 2 changed from %s, processing %v", skippedTemps, srcfg)
```
Comment (bparees) [inline pkg/stub/handler.go]: same comment on output

File: pkg/stub/handler.go [hunk 8]
```diff
@@ -933,11 +940,27 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
...
+	// delete entries that have been removed
+	for key, _ := range h.skippedImagestreams {
+		if _, ok := newStreamMap[key]; !ok {
+			delete(h.skippedImagestreams, key)
+		}
+	}
+	for key, _ := range h.skippedTemplates {
+		if _, ok := newTempMap[key]; !ok {
+			delete(h.skippedTemplates, key)
+		}
```
Comment (bparees) [inline pkg/stub/handler.go]: can't you just assign h.skippedImageStreams=newStreamMap and h.skippedTemplates=newTempMap ?

**Comment by gabemontero** [comment]
pushed updates @bparees  but also found an additional problem ... I'll highlight it in the file changed view momentarily

File: pkg/stub/handler.go [hunk 9]
```diff
@@ -299,12 +302,13 @@ func (h *Handler) StoreCurrentValidConfig(condition v1alpha1.SamplesResourceCond
 	cm.Data[regkey] = srcfg.Spec.SamplesRegistry
 	var value string
 	for _, val := range srcfg.Spec.SkippedImagestreams {
-		value = val + " "
+		value = value + val + " "
```
Comment (gabemontero) [inline pkg/stub/handler.go]: This was the additional fix

File: pkg/stub/handler.go [hunk 10]
```diff
@@ -299,12 +302,13 @@ func (h *Handler) StoreCurrentValidConfig(condition v1alpha1.SamplesResourceCond
...
 	}
 	cm.Data[skippedstreamskey] = value
+
 	value = ""
 	for _, val := range srcfg.Spec.SkippedTemplates {
-		value = val + " "
+		value = value + val + " "
```
Comment (gabemontero) [inline pkg/stub/handler.go]: This was the additional fix

**Pull request merged**